tests/test_beams.py

```diff
@@ -188,3 +287,95 @@ def test_beam_interpolation_pol():
         beam_analytic, freq, n_pix_lm=200, polarized=False
     )
     assert len(beam_pix.shape) == len(beam_pix_unpol.shape) + 2
+
+
+def test_polarized_not_efield(beam_list_unpol, crd_eq, eq2tops, sky_flux, freq, antpos):
+    """Test that when doing polarized sim, error is raised if beams aren't efield."""
+    with pytest.raises(ValueError) as e:
+        vis_cpu(
+            antpos,
+            freq[0],
+            eq2tops,
+            crd_eq,
+            sky_flux[:, 0],
+            beam_list=beam_list_unpol,
+            precision=2,
+            polarized=True,
+        )
+
+        assert "beam type must be efield" in str(e)
```

Comment:
you can make this shorter if you use the "match" parameter in `pytest.raises` like this:
```suggestion
    with pytest.raises(ValueError, match="beam type must be efield"):
        vis_cpu(
            antpos,
            freq[0],
            eq2tops,
            crd_eq,
            sky_flux[:, 0],
            beam_list=beam_list_unpol,
            precision=2,
            polarized=True,
        )
```
And similarly in other examples below.

Reply:
Ah, didn't know this! Nice :-)